Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -1207,55 +1207,97 @@ uint8_t AODProducerWorkflowDPL::getTRDPattern(const o2::trd::TrackTRD& track)
   return pattern;
 }
 
+template <typename TCaloHandler, typename TCaloCursor, typename TCaloTRGCursor>
+void AODProducerWorkflowDPL::addToCaloTable(const TCaloHandler& caloHandler, const TCaloCursor& caloCellCursor, const TCaloTRGCursor& caloTRGCursor,
+                                            int eventID, int bcID, int8_t caloType)
+{
+  auto inputEvent = caloHandler.buildEvent(eventID);
+  auto cellsInEvent = inputEvent.mCells; // get cells belonging to current event
+  for (auto& cell : cellsInEvent) {      // loop over all cells in collision
+    caloCellCursor(0,
+                   bcID,
+                   CellHelper::getCellNumber(cell),
+                   truncateFloatFraction(CellHelper::getAmplitude(cell), mCaloAmp),
+                   truncateFloatFraction(CellHelper::getTimeStamp(cell), mCaloTime),
+                   cell.getType(),
+                   caloType); // 1 = emcal, -1 = undefined, 0 = phos
+
+    // todo: fix dummy values in CellHelper when it is clear what is filled for trigger information
+    if (CellHelper::isTRU(cell)) { // Write only trigger cells into this table
+      caloTRGCursor(0,
+                    bcID,
+                    CellHelper::getFastOrAbsID(cell),
+                    CellHelper::getLnAmplitude(cell),
+                    CellHelper::getTriggerBits(cell),
+                    caloType);
+    }
+  }
+}
+
 // fill calo related tables (cells and calotrigger table)
-template <typename TEventHandler, typename TCaloCells, typename TCaloTriggerRecord, typename TCaloCursor, typename TCaloTRGTableCursor>
-void AODProducerWorkflowDPL::fillCaloTable(TEventHandler* caloEventHandler, const TCaloCells& calocells, const TCaloTriggerRecord& caloCellTRGR,
-                                           const TCaloCursor& caloCellCursor, const TCaloTRGTableCursor& caloCellTRGTableCursor,
-                                           const std::map<uint64_t, int>& bcsMap, int8_t caloType)
+template <typename TCaloCursor, typename TCaloTRGCursor>
+void AODProducerWorkflowDPL::fillCaloTable(const TCaloCursor& caloCellCursor, const TCaloTRGCursor& caloTRGCursor,
+                                           const std::map<uint64_t, int>& bcsMap,
+                                           const o2::globaltracking::RecoContainer& data)
 {
-  uint64_t globalBC = 0;    // global BC ID
-  uint64_t globalBCRel = 0; // BC id reltive to minGlBC (from FIT)
+  // get calo information
+  auto caloEMCCells = data.getEMCALCells();
+  auto caloEMCCellsTRGR = data.getEMCALTriggers();
+
+  auto caloPHOSCells = data.getPHOSCells();
+  auto caloPHOSCellsTRGR = data.getPHOSTriggers();
```

Comment:
Look fine, just 1 comment: in this way, if at least one of the calorimeters are requested, both will be filled. To avoid this you can add here 
```
if (!mInputSources[GIndex::PHS]) {
  caloPHOSCells = {};
  caloPHOSCellsTRGR = {};
}
if (!mInputSources[GIndex::EMC]) {
  caloEMCCells = {};
  caloEMCCellsTRGR = {};
}

Reply:
Thank you for the comment. I have added these checks after L1248